endpoint-agents/src/test/java/com/thousandeyes/sdk/endpoint/agents/EndpointAgentsApiTest.java

```diff
@@ -0,0 +1,1638 @@
+import com.thousandeyes.sdk.endpoint.agents.model.EndpointAgentUpdate;
+import com.thousandeyes.sdk.endpoint.agents.model.Error;
+import com.thousandeyes.sdk.endpoint.agents.model.ExpandEndpointAgentOptions;
+import com.thousandeyes.sdk.endpoint.agents.model.FilterEndpointAgentsResponse;
+import com.thousandeyes.sdk.endpoint.agents.model.ListEndpointAgentsResponse;
+import java.util.UUID;
+import com.thousandeyes.sdk.endpoint.agents.model.UnauthorizedError;
+import com.thousandeyes.sdk.endpoint.agents.model.ValidationError;
+import static com.github.tomakehurst.wiremock.client.WireMock.*;
+import static com.github.tomakehurst.wiremock.common.ContentTypes.AUTHORIZATION;
+import static com.github.tomakehurst.wiremock.common.ContentTypes.CONTENT_TYPE;
+import static com.thousandeyes.sdk.serialization.JSON.getDefault;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.github.tomakehurst.wiremock.junit5.WireMockRuntimeInfo;
+import com.github.tomakehurst.wiremock.junit5.WireMockTest;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import com.thousandeyes.sdk.client.ApiClient;
+import com.thousandeyes.sdk.client.ApiException;
+import com.thousandeyes.sdk.client.NativeApiClient;
+
+
+/**
+ * Request and Response model deserialization tests for EndpointAgentsApi
+ */
+@WireMockTest
+public class EndpointAgentsApiTest {
+    private static final String TOKEN = "valid-token";
+    private static final String BEARER_TOKEN = "Bearer %s".formatted(TOKEN);
+    private static EndpointAgentsApi api;
+    private final ObjectMapper mapper = getDefault()
+            .getMapper()
+            .configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, true);
+
+    @BeforeAll
+    public static void setup(WireMockRuntimeInfo wireMockRuntimeInfo) {
+        ApiClient client = NativeApiClient.builder()
+                                .baseUri(wireMockRuntimeInfo.getHttpBaseUrl())
+                                .bearerToken(TOKEN)
+                                .build();
+        api = new EndpointAgentsApi(client);
+    }
+    
+    /**
+     * Delete endpoint agent
+     * <p>
+     * Deletes the agent with the specified &#x60;agent_id&#x60;. 
+     *
+     * @throws JsonProcessingException if the deserialization fails
+     */
+    @Test
+    public void deleteEndpointAgentRequestAndResponseDeserializationTest()
+            throws JsonProcessingException, ApiException
+    {
+        UUID agentId = UUID.fromString("861b7557-cd57-4bbb-b648-00bddf88ef49");
```

Comment:
Sometimes id type will be UUID, and this is actually the example coming from the generator, with code and everything